zstacklib/zstacklib/utils/upload_task.py

```
@@ -0,0 +1,355 @@
+
+        slice_downloaded_size = 0
+        remaining = param.slice_size
+        bytes_read = 0
+        md5 = hashlib.md5()
+        chunks = []
+        chunk_size = 32 * 1024
+        try:
+            reader = SizedReader(p.fp, None, param.slice_offset)
+            while remaining > 0:
+                if task.lastError:
+                    raise Exception(task.lastError)
```

Ruff (0.13.1)

Create your own exception

(TRY002)